runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/util/TestPlatformProvider.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.util
+
+/**
+ * An implementation of [PlatformProvider] meant for testing
+ * @param env Environment variable mappings
+ * @param props System property mappings
+ * @param fs Filesystem path to content mappings
+ * @param os Operating system info to emulate
+ */
+public class TestPlatformProvider(
```

Comment:
I tried this but realized it would create a circular dependency between **:runtime-core** and **:testing** because `TestPlatformProvider` needs `FileSystem`, `OperatingSystem`, and a few other things in **:runtime-core**

Reply:
Oh you're right. Well then let's make it `@InternalApi`.